File: tier1/python/fd991c0f/fetcher.py
from urllib.parse import urlparse

import requests

PERMITTED = {"api.example.com", "cdn.example.com"}


def body(target):
    parsed = urlparse(target)
    if parsed.scheme != "https" or parsed.hostname not in PERMITTED:
        raise ValueError(target)
    response = requests.get(target, timeout=5)
    return response.text

File: tier1/javascript/40ea4ff9/fetcher.js
const PERMITTED = new Set(['api.example.com', 'cdn.example.com']);

async function body(target) {
  const parsed = new URL(target);
  if (parsed.protocol !== 'https:' || !PERMITTED.has(parsed.hostname)) {
    throw new Error('rejected');
  }
  const response = await fetch(target);
  return response.text();
}

module.exports = { body };
